src/ERC1967Ownable.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.7;
+
+import {ERC1967Upgrade} from "openzeppelin-contracts/proxy/ERC1967/ERC1967Upgrade.sol";
+
+/// @notice The proxy-safe ownability mix-in.
+/// It's supposed to be used only in contracts behind a proxy.
+///
+/// This contract uses the ERC-1967 admin slot to store the owner address.
+/// While this contract is capable of updating the owner,
+/// the proxy is expected to set up the initial value of the ERC-1967 admin.
+/// All instances of `ERC1967Ownable` contracts have owner set and locked to `0x00`.
+abstract contract ERC1967Ownable is ERC1967Upgrade {
```

Comment:
Open Zeppelin doesn't offer an upgradable ownership contract, i assume?

Reply:
They don't, they only have simple ERC-1967 `admin` functions, which are rather bare bones. The classic `Ownable` isn't suitable here, even if gutted, because it uses a storage slot and you can't stop it from doing so, because it writes to it in the constructor. It also has `renounceOwnership`, which doesn't make sense here.